syntax = "proto3";

package LLVM.TDG;

message DynamicLLVMValue {
  oneof value {
    uint64 v_int = 1;
    bytes v_bytes = 2;
  }
}

message DynamicLLVMInstruction {
  uint64 uid = 1;
  repeated DynamicLLVMValue params = 2;
  DynamicLLVMValue result = 3;
}

// Logs information on entering a function.
message DynamicLLVMFunctionEnter {
  string func = 1;
  repeated DynamicLLVMValue params = 2;
}

message DynamicLLVMTraceEntry {
  oneof Entry {
    DynamicLLVMInstruction inst = 1;
    DynamicLLVMFunctionEnter func_enter = 2;
  }
}

